deps/Makefile

```diff
@@ -613,44 +665,35 @@ else
 UV_FLAGS = --disable-shared $(UV_MFLAGS)
 endif
 
-libuv/configure:
-	(cd .. && git submodule init && git submodule update)
-ifeq (exists, $(shell [ -d libuv/.git ] && echo exists ))
-libuv/config.status: libuv/.git/HEAD
-endif
-ifeq (exists, $(shell [ -d $(JULIAHOME)/.git/modules/deps/libuv ] && echo exists ))
-libuv/config.status: $(JULIAHOME)/.git/modules/deps/libuv/HEAD
-endif
-libuv/config.status: libuv/configure
-	cd libuv && \
+$(LIBUV_SRC_DIR)/config.status: $(LIBUV_SRC_DIR)/configure
+	touch -c $(LIBUV_SRC_DIR)/aclocal.m4 # touch a few files to prevent autogen from getting called
+	touch -c $(LIBUV_SRC_DIR)/Makefile.in
+	touch -c $(LIBUV_SRC_DIR)/configure
+	cd $(LIBUV_SRC_DIR) && \
 	./configure --with-pic $(CONFIGURE_COMMON) $(UV_FLAGS)
 	touch -c $@
-$(UV_SRC_TARGET): libuv/config.status
-	touch -c libuv/aclocal.m4
-	touch -c libuv/Makefile.in
-	touch -c libuv/configure
-	touch -c libuv/config.status
-	$(MAKE) -C libuv $(UV_MFLAGS)
+$(UV_SRC_TARGET): $(LIBUV_SRC_DIR)/config.status
+	$(MAKE) -C $(LIBUV_SRC_DIR) $(UV_MFLAGS)
 	touch -c $@
-libuv/checked: $(UV_SRC_TARGET)
+$(LIBUV_SRC_DIR)/checked: $(UV_SRC_TARGET)
 ifeq ($(OS),$(BUILD_OS))
-	-$(MAKE) -C libuv check
+	-$(MAKE) -C $(LIBUV_SRC_DIR) check
 endif
 	echo 1 > $@
 $(UV_OBJ_TARGET): $(UV_SRC_TARGET)
-	$(MAKE) -C libuv install $(MAKE_COMMON)
+	$(MAKE) -C $(LIBUV_SRC_DIR) install $(MAKE_COMMON)
 	$(INSTALL_NAME_CMD)libuv.$(SHLIB_EXT) $(build_shlibdir)/libuv.$(SHLIB_EXT)
 
 clean-uv:
-	-$(MAKE) -C libuv clean
+	-$(MAKE) -C $(LIBUV_SRC_DIR) clean
 	-rm -rf $(build_shlibdir)/libuv.a $(build_includedir)/uv.h $(build_includedir)/uv-private
-distclean-uv: clean-uv
-	-$(MAKE) -C libuv distclean
+distclean-uv:
+	-rm -rf $(LIBUV_SRC_DIR)
```

Comment:
right. i added the tarballs as an afterthought and forgot to add the rule for it. i was going to move these to the git-external section anyways though